### PR TITLE
update search and model

### PR DESCRIPTION
**Pull request by msladek**
https://synjira.atlassian.net/browse/CELDEV-870

### CODE REVIEW

**Comment by fpichler** [comment]
CELDEV-870 is the wrong issue for the JavaScript Event-Manager extension.

File: src/main/webapp/resources/celJS/initCelements.js [hunk 1]
```diff
@@ -677,7 +677,7 @@
         _me._className = className;
         _me._actionFunction = actionFunction;
         if (condition && !/\s|;/.test(condition)) {
-          _me._conditionFunction = new Function('me', 'return ' + condition + ';');
```
Comment (fpichler) [inline src/main/webapp/resources/celJS/initCelements.js]: changing the first parameter is a nogo. This is braking all existing implementations.
Furthermore I prefer to NOT extend the jsCondition for security reasons.

File: src/main/webapp/resources/celJS/initCelements.js [hunk 2]
```diff
@@ -693,17 +693,27 @@
   
       _actionHandler : function(event) {
         var _me = this;
-        $$(_me._cssSelector).each(function(htmlElement) {
-          if (!_me._conditionFunction || _me._conditionFunction(htmlElement)) {
-            _me._actionFunction(htmlElement, _me._className);
+        _me._getTargetElements().each(function(targetElement) {
+          if (!_me._conditionFunction || _me._conditionFunction(_me._htmlElement, targetElement)) {
```
Comment (fpichler) [inline src/main/webapp/resources/celJS/initCelements.js]: changing the first parameter is a nogo. This is braking all existing implementations. It is not backwards compatible.

**Review by fpichler — CHANGES_REQUESTED** [review]
add a proper CELDEV issue to document the improvement in the release notes. Remove the brakeing 'origin' change and keep the '^' enhancement of the css selector.

File: src/main/webapp/resources/celJS/initCelements.js (hunk 1 above)
Comment (msladek) [reply src/main/webapp/resources/celJS/initCelements.js]: I require the origin parameter for another issue. In what way does it raise the already existing security concern (which we have addressed by limiting the condition to a single statement) ?

File: src/main/webapp/resources/celJS/initCelements.js (hunk 2 above)
Comment (msladek) [reply src/main/webapp/resources/celJS/initCelements.js]: the only usage of `_conditionFunction` can only be on this line, where the parameter order is correct. the condition string uses `me` to reference the `targetElement`, hence it doesn't depend on the order.

**Review by fpichler — APPROVED** [review]

**Pull request merged**